File: oauth2/src/main/java/chatox/oauth2/config/PlatformConfig.java
package chatox.oauth2.config;

import chatox.platform.exception.metadata.ServletChatoxErrorAttributes;
import chatox.platform.log.logger.MethodExecutionLogger;
import chatox.platform.time.TimeService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class PlatformConfig {
    @Bean
    public TimeService timeService() {
        return new TimeService();
    }

    @Bean
    public ServletChatoxErrorAttributes errorAttributes() {
        return new ServletChatoxErrorAttributes();
    }

    @Bean
    public MethodExecutionLogger methodExecutionLogger() {
        return new MethodExecutionLogger();
    }
}

File: chat/src/main/kotlin/chatox/chat/config/PlatformConfig.kt
package chatox.chat.config

import chatox.platform.exception.metadata.ReactiveChatoxErrorAttributes
import chatox.platform.log.logger.MethodExecutionLogger
import chatox.platform.pagination.process.PaginationParametersProcessor
import chatox.platform.time.TimeService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class PlatformConfig {

    @Bean
    fun paginationParametersProcessor() = PaginationParametersProcessor()

    @Bean
    fun methodExecutionLogger() = MethodExecutionLogger()

    @Bean
    fun errorAttributes() = ReactiveChatoxErrorAttributes()

    @Bean
    fun timeService() = TimeService()
}
